### PR TITLE
ledger: restore pointer receivers and args from #4003

### PR DESCRIPTION
**Pull request by cce**
## Summary

In #4003 some pointer receivers and pointer arguments were changed to pass by value and value receivers, which could lead to a performance regression. This changes them back.

There is a small discussion, including a benchmark, in https://github.com/algorand/go-algorand/pull/4003/files#r894844990 (a comment thread on ledger/accountdb.go line 1438)

## Test Plan

Existing tests should pass

### CODE REVIEW

File: ledger/ledgercore/accountdata.go [hunk 1]
```diff
@@ -145,7 +143,7 @@ func (u AccountData) MinBalance(proto *config.ConsensusParams) (res basics.Micro
 
 // IsZero checks if an AccountData value is the same as its zero value.
 func (u AccountData) IsZero() bool {
-	return reflect.DeepEqual(u, AccountData{})
+	return u == AccountData{}
```
Comment (cce) [inline ledger/ledgercore/accountdata.go:146]: I also changed this — I believe this was introduced in https://github.com/algorand/go-algorand/pull/126/files#diff-831e406846493c27b22aca1b7482743df94cec8a6957b34059c912ff7996ec28R268 when asset maps were added to AccountData, but now that there are no maps in AccountData we can use regular equality against an empty structure. Or we could change this to a pointer receiver.

File: ledger/accountdb.go [hunk 2]
```diff
@@ -1517,12 +1517,17 @@ func (ba *baseAccountData) GetAccountData() basics.AccountData {
...
 // IsEmpty returns true if all of the fields other than UpdateRound are zero.
-func (bv baseVotingData) IsEmpty() bool {
-	return bv == baseVotingData{}
+func (bv *baseVotingData) IsEmpty() bool {
+	return bv.VoteID == crypto.OneTimeSignatureVerifier{} &&
+		bv.SelectionID == crypto.VRFVerifier{} &&
+		bv.VoteFirstValid == basics.Round(0) &&
+		bv.VoteLastValid == basics.Round(0) &&
+		bv.VoteKeyDilution == 0 &&
+		bv.StateProofID == merklesignature.Verifier{}
```
Comment (jannotti) [inline ledger/accountdb.go:1526]: I'm confused about what happened here. I recall the discussion, but don't recall how the plain equality check (that you've reverted here) ended up being used. It seems wrong, given the comment, so I'm in favor of this change.  But I wonder if callers were changed somehow to make the simple check correct?

I also wonder if Will's suggestion, which I liked a lot for staying correct in the face of baseVotingData changes, is fast enough to consider - maybe things changed with the pointer/value swap.  He had suggested we implement this as an `==` check against: `baseVotingData{UpdateRound: bv.UpdateRound}` which nicely shows we want to check everything except the UpdateRound, and needs no updates.

Comment (jannotti) [reply ledger/accountdb.go:1526]: oh, ok, then yes, we can just use the simple `==` here.

Comment (cce) [reply ledger/accountdb.go]: restored to old version

**Review by algorandskiy — APPROVED** [review]

**Review by jannotti — APPROVED** [review]

**Pull request merged**